### PR TITLE
chore: Adopt black formatting

### PR DESCRIPTION
**Pull request by jfuss**
*Issue #, if available:*
N/A

*Description of changes:*
Move this repo from flake8 to black. Previous to this, developers would run `make pr` and handle any pylint or flake8 errors reported manually. With black, which is similar to go's fmt, will automatically format all the files for you. I also added 'pre-commit` configuration, so that way developers can install the git hooks and will ensure all files were formatted with black before doing a push.

This will align this repo with AWS SAM CLI's and provide a consistent developer experience for contribution across these repos.

*Description of how you validated changes:*
Ran `make pr`

*Checklist:*

- [ ] Write/update tests
- [x] `make pr` passes
- [ ] Update documentation
- [ ] Verify transformed template deploys and application functions as expected
- [ ] Add/update example to `examples/2016-10-31`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ShreyaGangishetty** [comment]
@jfuss I have rebased this PR to  to merge into develop. Could you please resolve the merge conflicts?

**Comment by jfuss** [comment]
@ShreyaGangishetty Rebased with HEAD of develop.

On a side note: If the intention is for developers to put code into develop, you should consider moving the default branch to develop instead of master. I expect the default branch to be where code will end up.

**Comment by ShreyaGangishetty** [comment]
@jfuss Thank you for this suggestion! We had our official SAM documentation in Github until recently. We didn't want to redirect everyone to our develop documentation until the documentation is made official through releases. Now that we have [AWS SAM specification](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-specification.html) it is a good idea to make `develop` as the default branch

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+[tool.black]
+line-length = 120
+target_version = ['py27', 'py37', 'py36']
```
Comment (keetonian) [inline pyproject.toml]: Can we add `'py38'`?

File: samtranslator/model/exceptions.py [hunk 2]
```diff
@@ -23,17 +25,21 @@ class DuplicateLogicalIdException(Exception):
...
 
     @property
     def message(self):
-        return 'Transforming resource with id [{logical_id}] attempts to create a new' \
-               ' resource with id [{duplicate_id}] and type "{type}". A resource with that id already' \
-               ' exists within this template. Please use a different id for that resource.'.format(
-                   logical_id=self._logical_id, type=self._type, duplicate_id=self._duplicate_id)
+        return (
+            "Transforming resource with id [{logical_id}] attempts to create a new"
+            ' resource with id [{duplicate_id}] and type "{type}". A resource with that id already'
```
Comment (keetonian) [inline samtranslator/model/exceptions.py]: Did you mean to leave this `'` single quote here?

**Pull request merged**